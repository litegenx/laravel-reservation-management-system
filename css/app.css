.router-link-active {
  cursor: pointer;
}

