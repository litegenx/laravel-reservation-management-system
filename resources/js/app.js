require('./bootstrap');
import 'babel-polyfill';
import Vue from 'vue';
import router from './router';
import store from './store';
import App from './components/App';
import vuetify from './plugins/vuetify';

new Vue({
    el: '#app',
    store,
    router,
    components: { App },
    template: '<App/>',
    vuetify,
});
