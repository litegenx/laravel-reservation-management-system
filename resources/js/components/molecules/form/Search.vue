<template>
    <v-dialog
        v-model="dialog"
        width="400"
    >
        <template v-slot:activator="{ on }">
            <v-text-field
                :label="label"
                :value="display"
                :prepend-icon="icon"
                readonly
                v-on="on"
                :error-messages="validateErrors"
            />
        </template>
        <v-layout
            wrap
            justify-center
        >
            <v-card
                class="pa-6"
                width="100%"
            >
                <v-card-text>
                    <v-select
                        ref="select"
                        :disabled="isDisabledSelect"
                        :label="label"
                        :value="value"
                        :items="items"
                        :item-text="searchText"
                        :item-value="searchValue"
                    />
                    <div class="text-right">
                        <v-btn
                            :disabled="isDisabledSelect"
                            @click="buttonClicked"
                        >
                            {{ $t('misc.ok') }}
                        </v-btn>
                    </div>
                </v-card-text>
                <v-card-text>
                    <v-form @submit.prevent="searchItem">
                        <v-text-field
                            :disabled="isSearching"
                            v-model="searchWord"
                        />
                        <div class="text-right">
                            <v-btn
                                :disabled="isSearching"
                                type="submit"
                            >
                                {{ $t('misc.search') }}
                            </v-btn>
                        </div>
                    </v-form>
                </v-card-text>
            </v-card>
        </v-layout>
    </v-dialog>
</template>

<script>
    import { mapGetters, mapActions } from 'vuex';
    import { get } from 'lodash';
    import { arrayToObject } from '../../../utils/misc';

    export default {
        data () {
            return {
                dialog: false,
                searchWord: '',
                isSearching: false,
                items: [],
                selectedId: null,
                selectedItem: null,
            };
        },
        props: {
            detail: {
                type: Object,
                required: true,
            },
            formInputs: {
                type: Object,
                required: true,
            },
            form: {
                type: Object,
                required: true,
            },
            label: {
                type: String,
                required: true,
            },
            value: {
                type: [String, Number],
                required: true,
            },
            validateErrors: {
                type: Array,
                required: true,
            },
        },
        computed: {
            ...mapGetters({
                model: 'crud/getTargetModel',
                getModelIcon: 'getModelIcon',
            }),
            icon () {
                return this.getModelIcon(this.searchModel);
            },
            searchSettings () {
                return this.form.search.split(':');
            },
            searchModel () {
                return this.searchSettings[ 0 ];
            },
            searchValue () {
                return this.searchSettings[ 1 ];
            },
            searchText () {
                return this.searchSettings[ 2 ];
            },
            isDisabledSelect () {
                return this.isSearching || !this.items.length;
            },
            display () {
                if (this.selectedId === get(this.formInputs, this.form.name)) {
                    return this.selectedItem;
                }
                return get(this.detail, this.form.display);
            },
        },
        methods: {
            ...mapActions({
                search: 'crud/search',
            }),
            async searchItem () {
                if (this.isSearching) {
                    return;
                }
                if (!this.searchWord.trim()) {
                    return;
                }

                this.isSearching = true;
                this.items = await this.search({
                    model: this.searchModel,
                    query: {
                        s: this.searchWord.trim(),
                        count: 30,
                    },
                });
                this.isSearching = false;
            },
            buttonClicked () {
                if (this.$refs.select.selectedItems.length) {
                    this.selectedId = this.$refs.select.selectedItems[ 0 ].id;
                    this.selectedItem = get(arrayToObject(this.items, { getKey: ({ item }) => item.id })[ this.selectedId ], this.searchText);
                    this.$emit('input', this.selectedId);
                }
                this.dialog = false;
            },
        },
    };
</script>
