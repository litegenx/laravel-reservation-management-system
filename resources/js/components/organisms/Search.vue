<template>
    <v-overlay
        :value="isActiveOverlay"
        class="text-center"
    >
        <v-progress-circular
            indeterminate
            size="64"
            :color="progressColor"
        />
        <div
            v-if="message"
            class="mt-3 font-weight-bold title"
        >
            {{ $t(`loading.${message}`) }}
        </div>
    </v-overlay>
</template>

<script>
    import { mapGetters } from 'vuex';

    export default {
        computed: {
            ...mapGetters({
                isActiveOverlay: 'loading/isActiveOverlay',
                message: 'loading/getMessage',
                progressColor: 'loading/getProgressColor',
            }),
        },
    };
</script>
