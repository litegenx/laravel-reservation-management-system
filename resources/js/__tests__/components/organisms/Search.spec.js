import { mount } from '@vue/test-utils';
import Vuex from 'vuex';
import Search from '../../../components/organisms/Search';
import app from '../app';
import setupLocalVue from '../setupLocalVue';

jest.useFakeTimers();

describe('Search', () => {
    it('should render search', () => {
        const wrapper = mount(app(Search, 'inner'), setupLocalVue({
            store: new Vuex.Store({
                getters: {
                    'loading/isActiveOverlay': () => false,
                    'loading/getProgressColor': () => '',
                    'loading/getMessage': () => '',
                },
            }),
        }));
        expect(wrapper.element).toMatchSnapshot();

        expect(wrapper.isVueInstance()).toBeTruthy();
        expect(wrapper.findAll('circle')).toHaveLength(0);
    });
});
