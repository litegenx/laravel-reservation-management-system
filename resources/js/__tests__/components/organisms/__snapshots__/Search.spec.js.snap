// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render search 1`] = `
<div
  class="v-application v-application--is-ltr theme--light"
  data-app="true"
  id="app"
>
  <div
    class="v-application--wrap"
  >
    <main
      app="true"
      class="v-content"
      classname="mb-5"
      style="padding: 0px 0px 0px 0px;"
    >
      <div
        class="v-content__wrap"
      >
        <div
          class="v-overlay text-center theme--dark"
          style="z-index: 5;"
        >
          <div
            class="v-overlay__scrim"
            style="opacity: 0; background-color: rgb(33, 33, 33); border-color: #212121;"
          />
        </div>
        <div
          id="test-contents"
        >
          contents
        </div>
      </div>
    </main>
  </div>
</div>
`;
